bin/pkg/lock.ml

```diff
@@ -5,6 +5,66 @@ module Opam_repo = Dune_pkg.Opam_repo
 module Lock_dir = Dune_pkg.Lock_dir
 module Pin_stanza = Dune_pkg.Pin_stanza
 
+module Progress_indicator = struct
+  module Per_lockdir = struct
+    module State = struct
+      module Repository = Dune_pkg.Pkg_workspace.Repository
+
+      type t =
+        | Initializing
+        | Updating_repos of Repository.Name.t list
+        | Solving
+        | Done
+
+      let pp = function
+        | Initializing -> Pp.text "Initializing..."
+        | Updating_repos repo_names ->
+          Pp.textf
+            "Updating package repos %s..."
+            (List.map repo_names ~f:(fun repo_name ->
+               Repository.Name.to_string repo_name |> String.quoted)
+             |> String.enumerate_and)
+        | Solving -> Pp.text "Solving..."
+        | Done -> Pp.text "Done!"
+      ;;
+    end
+
+    type t =
+      { lockdir_path : Path.Source.t
+      ; state : State.t ref
+      }
+
+    let create lockdir_path = { lockdir_path; state = ref State.Initializing }
+
+    let pp { lockdir_path; state } =
+      Pp.concat
+        [ Pp.textf "Locking %s: " (Path.Source.to_string lockdir_path |> String.quoted)
```

Comment:
The progress displays `"dune.lock"` with double quotes, when it shows the solution it says `Solution for dune.lock`, so, without the quotes. Either way is fine but I think we should be consistent. Maybe have a `pp` for lockdir names.

Reply:
I removed the quotes for consistency.